Content.Client/Shipyard/BUI/ShipyardConsoleBoundUserInterface.cs

```diff
@@ -0,0 +1,91 @@
+using Content.Client.Shipyard.UI;
+using Content.Client.Shipyard.Components;
+using Content.Shared.Shipyard.BUI;
+using Content.Shared.Shipyard.Events;
+using Robust.Client.GameObjects;
+using Robust.Client.Player;
+using Robust.Shared.Utility;
+using Robust.Shared.Prototypes;
+using static Content.Shared.Shipyard.Components.SharedShipyardConsoleComponent;
+using static Robust.Client.UserInterface.Controls.BaseButton;
+
+namespace Content.Client.Shipyard.BUI
+{
+    public sealed class ShipyardConsoleBoundUserInterface : BoundUserInterface
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+        [ViewVariables]
+        private ShipyardConsoleMenu? _menu;
+
+        [ViewVariables]
+        public int Balance { get; private set; }
+
+        public ShipyardConsoleBoundUserInterface(ClientUserInterfaceComponent owner, Enum uiKey) : base(owner, uiKey)
+        {
+        }
+
+        protected override void Open()
+        {
+            base.Open();
+            //We are bringing the sprite manager along for the future so we can flair up the menu with some icons later too, im just bad at UI design
+            var sysManager = _entityManager.EntitySysManager;
+            var spriteSystem = sysManager.GetEntitySystem<SpriteSystem>();
+            _menu = new ShipyardConsoleMenu(this, IoCManager.Resolve<IPrototypeManager>(), spriteSystem);
```

Comment:
I've been told to avoid using IoCManager if at all possible but I don't understand why exactly, so I pass the managers in manually

Reply:
Well you're calling IoCManager.Resolve in the first place in what you're originally doing, and my ForensicScannerMenu code was merged upstream without complaint. A quick grep for `InjectDependencies` reveals dozens of `xaml.cs` files that also do it this way, so there's precedent for it. If it's wrong to do it, it sure is prevalent.

> I've been told to avoid using IoCManager if at all possible but I don't understand why exactly, so I pass the managers in manually

Beware advice without explanation.

Reply:
Whenever I follow current implementations of things I get yelled at and told I should know better than to just do what everyone else does, I should be the one to know and do things correct lol.  I can change this later tonight.